### PR TITLE
Fix pylint error

### PR DESCRIPTION
**Pull request by jgfouca**
Test suite: code-checker (py3)
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review:

### CODE REVIEW

**Comment by jgfouca** [comment]
@jedwards4b , please re-review. I originally opened this PR solely for triggering a travis CI check. Their machine must have gotten some updates and the new pylint for py3 is a lot pickier. It found at least a dozen issues, some significant. I think I've fixed them.

File: scripts/lib/CIME/XML/batch.py [hunk 1]
```diff
@@ -93,10 +93,6 @@ def get_value(self, name, attribute=None, resolved=True, subgroup=None):
         if node is not None:
             value = self.text(node)
 
-        if value is None:
```
Comment (jgfouca) [inline scripts/lib/CIME/XML/batch.py]: This pattern was broken. The GenericXML.get_value is a placeholder that just always returns None.

File: scripts/lib/CIME/XML/generic_xml.py [hunk 2]
```diff
@@ -400,6 +400,8 @@ def set_value(self, vid, value, subgroup=None, ignore_type=True): # pylint: disa
         for node in valnodes:
             self.set_text(node, value)
 
+        return value if valnodes else None
```
Comment (jgfouca) [inline scripts/lib/CIME/XML/generic_xml.py]: This was a serious error. All of the subclasses of GenericXML return something if the set_value was successful, but this one wasn't doing that. Some of the subclasses were invoking this version and so this was a bug.

**Comment by jedwards4b** [comment]
There still appears to be a failure.

**Comment by jgfouca** [comment]
@jedwards4b , yes, a typo in one of my disables

**Comment by jgfouca** [comment]
As a side note, I want to thank the folks who pushed for py3. It looks like py3 and py3 libraries have moved, and are continuing to move, ahead of the py2 versions.

**Comment by mfdeakin-sandia** [comment]
Thanks again for actually implementing python 3 @jedwards4b !
As a side note, when I set the scream project up to use codacity for static analysis code reviews, it picked up a lot of potential issues in cime that you all might be interested in

**Comment by jedwards4b** [comment]
@mfdeakin-sandia Sure we would like to see them if you have an easy way to share.

**Comment by mfdeakin-sandia** [comment]
I did have them easily available, but don't know how to get to them since I excluded the CIME directory from the repo. Codacy is free for open source projects and trivial to set up with github, so you should be able to set it up for CIME itself.

**Pull request merged**